x-pack/packetbeat/magefile.go

```diff
@@ -144,9 +150,40 @@ func TestPackages() error {
 }
 
 func TestSystem(ctx context.Context) error {
-	mg.Deps(devtools.BuildSystemTestBinary)
+	mg.SerialDeps(getNpcapInstaller, devtools.BuildSystemTestBinary)
 
 	args := devtools.DefaultGoTestIntegrationArgs()
 	args.Packages = []string{"./tests/system/..."}
 	return devtools.GoTest(ctx, args)
 }
+
+// getNpcapInstaller gets the installer from the Google Cloud Storage service.
+//
+// On Windows platforms, if getNpcapInstaller is invoked with the environment variables
+// CI or NPCAP_LOCAL set to "true" and the OEM Npcap installer is not available it is
+// obtained from the cloud storage. This behaviour requires access to the private image.
+// If NPCAP_LOCAL is set to "true" and the file is in the npcap/installer directory, no
+// fetch will be made.
+func getNpcapInstaller() error {
+	// TODO: Consider whether to expose this as a target.
+	if runtime.GOOS != "windows" {
+		return nil
+	}
+	if os.Getenv("CI") != "true" && os.Getenv("NPCAP_LOCAL") != "true" {
+		return errors.New("only available if running in the CI or with NPCAP_LOCAL=true")
+	}
+	dstPath := filepath.Join("./npcap/installer", installer)
+	if os.Getenv("NPCAP_LOCAL") == "true" {
+		fi, err := os.Stat(dstPath)
+		if err == nil && !fi.IsDir() {
+			fmt.Println("using local Npcap installer with NPCAP_LOCAL=true")
+			return nil
+		}
+		if !errors.Is(err, fs.ErrNotExist) {
+			return err
+		}
+	}
+
+	fmt.Printf("getting %s from private cache\n", installer)
+	return sh.RunV("gsutil", "cp", "gs://obs-ci-cache/private/"+installer, dstPath)
```

Comment:
By using docker instead, then 

```
id=$(docker create docker.elastic.co/observability-ci/golang-crossbuild:1.16.9-npcap-1.60-debian8)
Unable to find image 'docker.elastic.co/observability-ci/golang-crossbuild:1.16.9-npcap-1.60-debian8' locally
1.16.9-npcap-1.60-debian8: Pulling from observability-ci/golang-crossbuild
b82b9923b08d: Already exists
a6b5b4985e67: Pulling fs layer

docker cp $id:/installer/npcap-1.60-oem.exe - > npcap-1.60-oem.exe
ls -l npcap-1.60-oem.exe 
-rw-r--r-- 1 vmartinez staff 1094144 Mar 21 10:16 npcap-1.60-oem.exe
docker rm $id
```

or
```
docker cp $(docker create --name npcap-1.60 --rm docker.elastic.co/observability-ci/golang-crossbuild:1.16.9-npcap-1.60-debian8):/installer/npcap-1.60-oem.exe .
Unable to find image 'docker.elastic.co/observability-ci/golang-crossbuild:1.16.9-npcap-1.60-debian8' locally
1.16.9-npcap-1.60-debian8: Pulling from observability-ci/golang-crossbuild
b82b9923b08d: Already exists
a6b5b4985e67: Pulling fs layer
...
Status: Downloaded newer image for docker.elastic.co/observability-ci/golang-crossbuild:1.16.9-npcap-1.60-debian8
ls -l npcap-1.60-oem.exe                                                                                                                      
-rw-r--r-- 1 vmartinez staff 1092304 Mar  7 04:04 npcap-1.60-oem.exe
docker rm npcap-1.60
```

Although, it might not work for windows :/

Reply:
>Although, it might not work for windows :/

This is my concern.